kotlin/internal/utils.bzl

```diff
@@ -187,11 +187,17 @@ def _maybe_make_srcsjar_action(ctx, srcs):
 
 
 # PACKAGE JARS #################################################################################################################################################
-def _fold_jars_action(ctx, output_jar, input_jars):
-    args=["--output", output_jar.path]
+def _fold_jars_action(ctx, rule_kind, output_jar, input_jars):
+    args=[
+        "--normalize",
+        "--compression",
+        "--deploy_manifest_lines",
```

Comment:
manually stamp this into the jar -- perhaps we need a better `stamp_jar` sidekick that merges and stamps ? .

Reply:
Another question about me manually stamping here and in the builder (using the JavaCreator). Will the "Target-Label" be picked up if I do it this way ? Skylark hasn't been made aware of it explicitly -- i.e., will Skylark read the manifest.

Reply:
> Is there a way inheriting the manifest from the primary jar with singlejar ?

Not currently. For the Java rules singlejar doesn't process the interface jars that are used at compile-time, which is the only time we need `Target-Label`.

> Skylark hasn't been made aware of it explicitly -- i.e., is Bazel meant to actively read the manifest for the stamp data ?.

Tools like JavaBuilder read the manifest data when they are executed. Bazel itself doesn't read it during analysis or anything.